src/JustSaying/Messaging/Middleware/ExactlyOnce/ExactlyOnceMiddleware.cs

```diff
@@ -1,38 +1,38 @@
 using System;
+using System.Threading;
 using System.Threading.Tasks;
-using JustSaying.Models;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Messaging.Middleware.Handle;
 using Microsoft.Extensions.Logging;
 
-namespace JustSaying.Messaging.MessageHandling
+namespace JustSaying.Messaging.Middleware.ExactlyOnce
```

Comment:
Does this need to live in its own namespace? Seems a bit OTT for one class and one extension method to me.

Reply:
You might be right - I was thinking that it'd be good to keep these separate as we add more (e.g. a polly retry/rate limiting), but having separate namespaces is maybe a bit unnecessary.

I've moved everything into a base `JustSaying.Messaging.Middleware` namespace while retaining the folder structure to keep things tidy, which makes things a bit more discoverable as you noted below.

What do you think?